### ISSUE TITLE
CS8762 should not warn here

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn#44285]
```cs
class C
{
    bool M([NotNullWhen(true)] out string? p)
    {
        p = null;
        // ↓ CS8762	Parameter 'p' may not have a null value when exiting with 'true'
        return p != null;
    }
}
```

**Comment by RikkiGibson** [comment]
I can't reproduce this in fa4ffc8.
